app/controlplane/internal/authz/authz_test.go

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Authorization package
+package authz
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddPolicies(t *testing.T) {
+	testcases := []struct {
+		name     string
+		subject  *SubjectAPIToken
+		policies []*Policy
+		wantErr  bool
+	}{
+		{
+			name:    "empty policies",
+			wantErr: true,
+		},
+		{
+			name: "no subject",
+			policies: []*Policy{
+				PolicyWorkflowContractList,
+			},
+			wantErr: true,
+		},
+		{
+			name: "happy path",
+			subject: &SubjectAPIToken{
+				ID: uuid.NewString(),
+			},
+			policies: []*Policy{
+				PolicyWorkflowContractList,
+				PolicyReferrerRead,
+			},
+		},
+		{
+			name: "another happy path",
```

Comment:
it could be nicer to state what path these are. "happy path" and "another happy path" are not identifying the test cases

Reply:
You are right, this is sloppy. I'll fix it. 

Thanks!